### ISSUE TITLE
Transmission has bad performance because of debug assertions

### ISSUE THREAD

**Issue by Sjord** [osmc/osmc#459]
I experience performance problems when downloading torrents on my RPi3 on OSMC. I think the current build of transmission-daemon has debug assertions enabled, which degrade performance. Transmission should be built with the -DNDEBUG compile flag. 

For example, if TR_ENABLE_ASSERTS is true, transmission [will call assertArrayIsSortedAndUnique](https://github.com/transmission/transmission/blob/master/libtransmission/ptrarray.c#L164) often to check its internal data structure. TR_ENABLE_ASSERTS is defined in [tr-assert.h](https://github.com/transmission/transmission/blob/c11f2870fd18ff781ca06ce84b6d43541f3293dd/libtransmission/tr-assert.h#L22), depending on the value of NDEBUG.

This means you can check whether you have a debug build by grepping for assertArrayIsSortedAndUnique on the binary:

    $ grep assertArrayIsSortedAndUnique /usr/bin/transmission-daemon 
    Binary file /usr/bin/transmission-daemon matches

Also, `objdump --syms`, shows a list of symbols.

A binary built with NDEBUG does not have the assertArrayIsSortedAndUnique function. The transmission binaries from debian don't have this function and don't list symbols.

I am not sure what the correct way to build a release build is. I managed it with `export CPPFLAGS=-DNDEBUG` and so on, but there should probably be a cmake option for this. I'll look into this.

**Comment by samnazarko** [comment]
Are you using the OSMC version or Debian version?

**Comment by Sjord** [comment]
I think I am using the OSMC version.

    $ dpkg -l transmission*
    dpkg-query: no packages found matching transmission*

The correct way to make build a release seems to be this command:

    cmake -DCMAKE_BUILD_TYPE=Release ..

**Comment by samnazarko** [comment]
Check for armv7-transmission-app-osmc.

**Comment by Sjord** [comment]
Yes, I have that.

```
$ dpkg -l armv7-transmission-app-osmc
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                                    Version                  Architecture             Description
+++-=======================================-========================-========================-===================================================================================
ii  armv7-transmission-app-osmc             2.9.3-1                  armhf                    Transmission for OSMC
```

I tested the debug flags by compiling transmission from git, but of course the .tar.xz distribution works differently and has different compile flags. I'll try to find the correct flags there.

**Comment by Sjord** [comment]
I am getting less and less sure that the binary has debug assertions enabled. Maybe my `grep` command is not a reliable indication whether assertions are enabled in a binary.

**Comment by Sjord** [comment]
Unlike the git version, the transmission-2.93.tar.xz distribution already comes with `CPPFLAGS=-DNDEBUG` enabled by default. So it seems it always builds a release binary, which makes sense. I tried it with the same flags OSMC uses to compile it, but it would always create a release build.

The grep and objdump commands I used don't seem to be reliable indicators for debug builds. A better indicator may be the function `checkBestScoresComeFirst`, since that won't be defined or called in a release build. The OSMC transmission-daemon binary doesn't contain this function, and that would indicate it is a release build.

So it seems I have been mistaken. Sorry to waste your time, @samnazarko, and thank you for the quick response.
